src/Discord/Discord.php

```diff
@@ -1486,9 +1486,9 @@ public function __set(string $name, $value)
      *
      * @param string|int $channel_id Id of the channel.
      *
-     * @return Channel|null
+     * @return Channel|Thread|null
      */
-    public function getChannel($channel_id): ?Channel
+    public function getChannel($channel_id): Channel|Thread|null
```

Comment:
```suggestion
    public function getChannel($channel_id): ?Part
```

Union types were only introduced in PHP 8.0 but we need to support PHP 7.4.

Reply:
I think I know what he mean.. in API docs it is mentioned that [Get Channel](https://discord.com/developers/docs/resources/channel#get-channel) may include a thread member object if channel is a thread.

```suggestion
public function getChannel($channel_id): ?Part
{
    foreach ($this->guilds as $guild) {
        foreach ($guild->channels as $channel) {
            if ($channel->id == $channel_id) {
                return $channel;
            } else if ($thread = $channel->threads->get('id', $channel_id)) {
                return $thread;
            }
        }
    }

    if ($channel = $this->private_channels->get('id', $channel_id)) {
        return $channel;
    }

    return null;
}
```
Not sure if the nested loop is overkill